modules/javafx.controls/src/test/java/test/javafx/scene/control/ButtonTest.java

```diff
@@ -420,9 +415,13 @@ private void show() {
         btn.fireEvent(new ActionEvent());
         btn.fire();
 
+        MouseEventFirer mouse = new MouseEventFirer(btn);
+
         mouse.fireMousePressed();
         mouse.fireMouseReleased();
         mouse.fireMouseClicked();
+
+        mouse.dispose();
     }
```

Comment:
just curious: do we have to dispose the mouseFirer? If so, that pattern isn't safe because it will not happen if the test fails.

Reply:
Hmm.. very good point. I believe, we should use dispose mechanism whenever available for a readable cleanup. Both classes MouseEventFirer & StageLoader provide dispose method and the user of these classes should use it.
Now, this can be handled in two ways -
1) Keep a reference at test class level - it will be null by default - individual tests will create the object and update the class reference - we can dispose in cleanup method after the test
2) Use Try blocks in tests that use MouseEventFirerer or StageLoader objects and dispose them in finally{} blocks.

We can use (1) if many tests in a test class use MouseEventFirerer or StageLoader objects - for example ComboBoxTest and use approach (2) if very few tests use these classes - for example ButtonTest.

Reply:
makes sense :)

Reply:
just looked at MouseEventFirer (a bit late, sry ;): all it does is to dispose the Stageloader it created (if it received a bare node) - so if we add a node to a stage and/or wrap it into our own loader, we don't really need to dispose the firer (though it doesn't hurt except a very minimal code clutter)

Reply:
If we use our own loader - we would need to dispose that as well :)

Also, I found approach (1) of using reference at test class level and cleaning it up in cleanup method after the test is more suitable as in ButtonTest, the assertion happens in event handler. Wrapping it in try..finally will need additional code.
I will push the changes in a moment.

Reply:
nit-picking: we already do dispose our own loader (or cleanup our stage), don't we :) 

the change looks good

Reply:
Can you please mark it as Approved?